### PR TITLE
Adding net http instrumentation

### PR DESCRIPTION
**Pull request by enoodle**
#### net/http
Wrapping http requests using the net/http library can be done by wraping the client:
```go
import (
	"github.com/epsagon/epsagon-go/wrappers/net/http"
...
	client := epsagonhttp.Wrap(http.Client{})
	resp, err := client.Get(anyurl)
```

Later we should find a way to wrap `http.DefaultClient` from the lambda wrapper without causing cyclic imports.

### CODE REVIEW

File: wrappers/net/http/client.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+	if err != nil {
+		errorcode = protocol.ErrorCode_ERROR
+	}
+	return &protocol.Event{
+		Id:        "http.Client-" + uuid.NewV4().String(),
+		Origin:    "http.Client",
+		ErrorCode: errorcode,
+		Resource: &protocol.Resource{
+			Name:      url,
+			Type:      "requests",
```
Comment (ranrib) [inline wrappers/net/http/client.go]: requests -> http

File: wrappers/net/http/client_test.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
+package epsagonhttp
+
+import (
+	// "fmt"
```
Comment (ranrib) [inline wrappers/net/http/client_test.go]: remove line?

File: wrappers/net/http/client.go (hunk 1 above)
Comment (enoodle) [reply wrappers/net/http/client.go]: :+1:

File: wrappers/net/http/client_test.go (hunk 2 above)
Comment (enoodle) [reply wrappers/net/http/client_test.go]: :+1:

File: example/simple_lambda/main.go [hunk 3]
```diff
@@ -4,15 +4,30 @@ import (
 	"github.com/aws/aws-lambda-go/events"
 	"github.com/aws/aws-lambda-go/lambda"
 	"github.com/epsagon/epsagon-go/epsagon"
+	"github.com/epsagon/epsagon-go/wrappers/net/http"
+	"io/ioutil"
```
Comment (henprtz) [inline example/simple_lambda/main.go]: Please rename this file to what it is - api_gateway_http

**Pull request merged**